lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -277,9 +276,39 @@ func (c *Client) start() {
 
 		// C.read_rdp_output blocks for the duration of the RDP connection and
 		// calls handle_bitmap repeatedly with the incoming bitmaps.
-		if errCode := C.read_rdp_output(c.rustClient); errCode != C.ErrCodeSuccess {
-			c.cfg.Log.Warningf("Failed reading RDP output frame: %v", errCode)
+		res := C.read_rdp_output(c.rustClient)
+
+		// Copy the returned message and free the C memory)
+		userMessage := C.GoString(res.user_message)
+		C.free_c_string(res.user_message)
+
+		// If the disconnect was initiated by the server or for
+		// an unknown reason, try to alert the user as to why via
+		// a TDP error message.
+		if res.disconnect_code != C.DisconnectCodeClient {
+			if err := c.cfg.Conn.WriteMessage(tdp.Error{
+				Message: fmt.Sprintf("The Windows Desktop disconnected: %v", userMessage),
+			}); err != nil {
+				c.cfg.Log.WithError(err).Error("error sending server disconnect reason over TDP")
```

Comment:
In this case, since the disconnect reason never made it to the user, we should probably log it.

Reply:
It will get logged in the section below